core/queryparser/sparql/src/main/java/org/eclipse/rdf4j/query/parser/sparql/SPARQLQueries.java

```diff
@@ -0,0 +1,131 @@
+/******************************************************************************* 
+ * Copyright (c) 2021 Eclipse RDF4J contributors. 
+ * All rights reserved. This program and the accompanying materials 
+ * are made available under the terms of the Eclipse Distribution License v1.0 
+ * which accompanies this distribution, and is available at 
+ * http://www.eclipse.org/org/documents/edl-v10.php. 
+ *******************************************************************************/
+package org.eclipse.rdf4j.query.parser.sparql;
+
+import org.eclipse.rdf4j.common.text.StringUtil;
+import org.eclipse.rdf4j.model.Namespace;
+
+/**
+ * Utility functions for working with SPARQL query strings.
+ * 
+ * @author Jeen Broekstra
+ */
+public class SPARQLQueries {
+
+	/**
+	 * Creates a string representing of the supplied {@link Namespace namespaces} as SPARQL prefix declarations. This
+	 * can be used when composing a SPARQL query string in code, for example:
+	 * 
+	 * <pre>
+	 * <code>
+	 * String query = SPARQLQueries.getPrefixClauses(connection.getNamespaces()) + "SELECT * WHERE { ?s ex:myprop ?o }";
```

Comment:
Looks nice. Would be even nicer if we could do something like `sail.prependNamespaces(true)` or something like that which would essentially do this automatically.

Reply:
Thought that is unrelated to this PR obviously. Just my wishful thinking :P

Reply:
There's some functionality already for automatically inserting "well-known" namespaces when preparing a SPARQL query, though it currently uses a hardcoded list of constants rather than whatever is available in the repository (this happens in the PrefixDeclProcessor which is part of the SPARQL parser). I wouldn't be against adding a configuration option at the level of the Repository API to automatically let it insert prefix declarations from the underlying store into SPARQL queries. As you said though: it's out of scope for this issue :)